### ISSUE TITLE
Native validation & accessibility

### ISSUE THREAD

**Issue by oliverjam** [foundersandcoders/mc-client-side-validation#21]
I think we should more strongly encourage students to use native HTML validation. Doing custom validation in JS and then just putting a message inside a `<p>` doesn't actually communicate semantically that an input is invalid.

The constraint validation API has the `inputField.setCustomValidity()` method to create custom validation errors that will actually set the field as invalid. So for example in the solution we would call `passwordInput.setCustomValidity("Your passwords need to match")` (and then again with an empty string when they do match).

There's [a good example of custom validation on MDN](https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/HTML5/Constraint_validation#Complex_constraints_using_the_Constraint_Validation_API)

We could also mention setting `aria-invalid="true"` in these cases, but setting custom validity should be enough
